src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -0,0 +1,1089 @@
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+namespace
+{
+    constexpr const char* VERT_SHADER_NO_UV_MAIN = 
+R"(
+void main()
+{
+    LocalPos = vPos;
+    gl_Position = irr_glsl_pseudoMul4x4with3x1(CamData.params.MVP, vPos);
+    ViewPos = irr_glsl_pseudoMul3x4with3x1(CamData.params.MV, vPos);
+    mat3 normalMat = irr_glsl_SBasicViewParameters_GetNormalMat(CamData.params);
+    Normal = normalMat*normalize(vNormal);
+}
+)";
+    constexpr const char* FRAG_SHADER_NO_UV_MAIN =
+R"(#define Ia 0.1
+void main()
+{
+    vec3 N = normalize(Normal);
+    vec3 L = normalize(-ViewPos);
+    vec3 R = -reflect(L,N);
+    float NdotL = max(dot(N,L), 0.0);
+    float VdotR = max(dot(L,R), 0.0);
+
+    vec3 color;
+    if (PC.params.extra&ILLUM_MODEL_MASK > 0)
+    {
+        color = PC.params.Ka*Ia + PC.params.Kd*NdotL;
+        switch (PC.params.extra&ILLUM_MODEL_MASK)
+        {
+        case 2:
+        case 3://2 + reflection map
+        case 4://3 with transparency (glass)
+        case 6:
+        case 8://reflection map
+        case 9://reflection map
+            color += PC.params.Ks*pow(VdotR, PC.params.Ns);
+            break;
+        case 5:
+        case 7:
+            color += PC.params.Ks*pow(VdotR, PC.params.Ns)*Fresnel_dielectric(PC.params.Ni, NdotL);
```

Comment:
always put the fresnel in (is the Titanfall fresnel renormalization factor applicable ? )

Reply:
+ do it as a proper mix, so the more fresnel weight on specular, the less diffuse (ambient is equally reflected by diffuse and specular, so fresnel won't affect it and it can be just "added" afterwards - don't make it part of diffuse)